decidim-admin/app/commands/decidim/admin/destroy_participatory_process_step.rb

```diff
@@ -30,7 +30,7 @@ def call
 
       def reorder_steps
         ReorderParticipatoryProcessSteps
-          .new(@participatory_process.steps, @participatory_process.steps.map(&:id))
+          .new(@participatory_process.steps.reload, @participatory_process.steps.map(&:id))
```

Comment:
I'd create a variable `steps = @participatory_process.steps.reload` just before calling the method and use it afterwards.

Also, it looks weird that you're reordering the steps using an arbitrary order that comes from the db, right?